vampyr/trees.cpp

```diff
@@ -0,0 +1,282 @@
+        .def("getNCoefs", &MWNode<D>::getNCoefs)      
+       // .def("getCoefs", py::overload_cast<const double *>(&MWNode<D>::getCoefs, py::const_))
+        .def("printCoefs", &MWNode<D>::printCoefs)
+        .def("zeroCoefs", &MWNode<D>::zeroCoefs)
+        .def("setCoefBlock", &MWNode<D>::setCoefBlock)
+        .def("addCoefBlock", &MWNode<D>::addCoefBlock)
+        .def("zeroCoefBlock", &MWNode<D>::zeroCoefBlock)
+        .def("calcNorms", &MWNode<D>::calcNorms)
+        .def("zeroNorms", &MWNode<D>::zeroNorms)
+        .def("clearNorms", &MWNode<D>::clearNorms)
+        .def("createChildren", &MWNode<D>::createChildren)
+        .def("genChildren", &MWNode<D>::genChildren)
+        .def("deleteChildren", &MWNode<D>::deleteChildren)
+        .def("cvTransform", &MWNode<D>::cvTransform)
+        .def("mwTransform", &MWNode<D>::mwTransform)
+        .def("splitCheck", &MWNode<D>::splitCheck)
+        .def("setHasCoefs", &MWNode<D>::setHasCoefs)
+        .def("setIsEndNode", &MWNode<D>::setIsEndNode)
+        .def("setIsGenNode", &MWNode<D>::setIsGenNode)
+        .def("setIsRootNode", &MWNode<D>::setIsRootNode)
+        .def("setIsLeafNode", &MWNode<D>::setIsLeafNode)
+        .def("setIsAllocated", &MWNode<D>::setIsAllocated)
+        .def("setIsBranchNode", &MWNode<D>::setIsBranchNode)
+        .def("setIsLooseNode", &MWNode<D>::setIsLooseNode)
+        .def("clearHasCoefs", &MWNode<D>::clearHasCoefs)
+        .def("clearIsEndNode", &MWNode<D>::clearIsEndNode)
+        .def("clearIsGenNode", &MWNode<D>::clearIsGenNode)
+        .def("clearIsRootNode", &MWNode<D>::clearIsRootNode)
+        .def("clearIsAllocated", &MWNode<D>::clearIsAllocated);
+
+
+   // py::class_<FunctionNode<D>> funcnode(m, "FunctionNode", mwnode);
+   // funcnode.def(py::init<MWNode<D>>());
+
+   // py::class_<GenNode<D>> gennode(m, "GenNode", funcnode);
+    //gennode.def(py::init<FunctionNode<D>>());
+
+
+    py::class_<HilbertPath<D>> hilbpath(m, "HilbertPath");
+    hilbpath.def(py::init<HilbertPath<D>>())
+        .def("getPath", &HilbertPath<D>::getPath)
+        .def("getChildPath", &HilbertPath<D>::getChildPath)
+        .def("getZIndex", &HilbertPath<D>::getZIndex)
+        .def("getHIndex", &HilbertPath<D>::getHIndex);
+
+
+
+    py::class_<NodeBox<D>> nodebox(m, "NodeBox");
+    nodebox.def(py::init<BoundingBox<D>>())
+       //.def("setNode", &NodeBox<D>::setNode);
+       //.def("setNode", py::overload_cast<int, MWNode<D>**>(&NodeBox<D>::setNode))
+        .def("clearNode", &NodeBox<D>::clearNode)      
+        .def("getNode", py::overload_cast<const NodeIndex<D> &>(&NodeBox<D>::getNode, py::const_))
+        .def("getNode", py::overload_cast<const Coord<D> &>(&NodeBox<D>::getNode, py::const_))
+        .def("getNode", py::overload_cast<int>(&NodeBox<D>::getNode, py::const_))
+        .def("getNOccupied", &NodeBox<D>::getNOccupied);
+
+
+
+    py::class_<NodeIndex<D>> nodeindex(m, "NodeIndex");
+    nodeindex.def(py::init<int, int *>())
```

Comment:
the `int *` wont work properly in vampyr. This should probably be changed to an `std::vector` in mrcpp. Add some comment on it saying it wont work since raw pointers does not exist in python